### ISSUE TITLE
CT Scan Preprocessing

### ISSUE THREAD

**Issue by xiaodaxia007** [StanfordMIMI/Merlin#10]
Hi, thanks for sharing the excellent work. I am playing with this model recently. In "CT Scan Preprocessing" section of the manuscript, it states that _we reformat all CT scans so that the first axis points from left to right, the second from posterior to anterior, and the third from inferior to superior._ I visualized the [image1.nii.gz file](https://huggingface.co/stanfordmimi/Merlin/tree/main). Does the first axis in your manuscript refer to the y axis of the figure and second axis refer to the x axis?

![Image](https://github.com/user-attachments/assets/0f6857ea-ae4e-4c29-ae7e-dcb08e1a0e5c)

I am a little bit confused since most of the research work feed the following slices into the deep learning model rather than rotating each slice by 90 degrees.

![Image](https://github.com/user-attachments/assets/31689843-3a7b-4dd7-be8d-3ddc599b489f)

Could you clarify the above point and also, could you explain why rotating 90 degree to train the ct-vlm? Your response will be very helpful. Thank you.

**Comment by ashwinkumargb** [comment]
That section is defining RAS orientation. It is a fairly standard preprocessing step for medical images to ensure that they are a common orientation before feeding it into a model. Did you try passing the image through the data loader? Just the nifti is the raw volume, but passed through the data loader (see below) should put you in the expected orientation. 

```python
ImageTransforms = Compose(
    [
        LoadImaged(keys=["image"]),
        EnsureChannelFirstd(keys=["image"]),
        Orientationd(keys=["image"], axcodes="RAS"),
        Spacingd(keys=["image"], pixdim=(1.5, 1.5, 3), mode=("bilinear")),
        ScaleIntensityRanged(
            keys=["image"], a_min=-1000, a_max=1000, b_min=0.0, b_max=1.0, clip=True
        ),
        SpatialPadd(keys=["image"], spatial_size=[224, 224, 160]),
        CenterSpatialCropd(
            roi_size=[224, 224, 160],
            keys=["image"],
        ),
        ToTensord(keys=["image"]),
    ]
)
```

**Comment by xiaodaxia007** [comment]
image1.nii.gz was visualized before dataloader:

![Image](https://github.com/user-attachments/assets/88959e61-7ee7-4fd2-9870-f7e87a5f50f4)


image1.nii.gz was visualized after dataloader:

![Image](https://github.com/user-attachments/assets/88029557-a61c-4c6a-8b86-20821021c308)

```
for batch in dataloader:
    
    print(batch["image"].shape)
    slices = batch["image"]
    print('slices:' + str(slices.shape))
    slices = slices[0,0]
    print('slices:' + str(slices.shape))
    ct_volume = slices.cpu().detach().numpy()
    print('slices:' + str(ct_volume.shape))

    # Function to plot a batch of slices
    def plot_slices(volume, start_index=0, num_slices=160):
        fig, axes = plt.subplots(16, 10, figsize=(20, 2 * 16))
        for i in range(num_slices):
            slice_idx = start_index + i
            if slice_idx >= volume.shape[2]:
                break
            ax = axes[i // 10, i % 10]
            ax.imshow(volume[:, :, slice_idx], cmap='gray')
            ax.set_title(f"Slice {slice_idx}")
            ax.axis('off')
        plt.tight_layout()
        plt.show()

    # Example: Show the first 20 slices
    plot_slices(ct_volume, start_index=0, num_slices=160)
    
    outputs = model(batch["image"].to(device), batch["text"])
    print("\n================== Output Shapes ==================")
    print(f"Contrastive image embeddings shape: {outputs[0].shape}")
    print(f"Phenotype predictions shape: {outputs[1].shape}")
    print(f"Contrastive text embeddings shape: {outputs[2].shape}") ```

To my understanding, they share the same orientation, but it does not follow the RAS convention. Could you clarify this? Thank you.
